### PR TITLE
Delay hiding of statusbar to avoid flicker

### LINKED ISSUE #234: statusbar should set a timer before hiding when item is removed
Currently with some of the LSP stuff we get a very blinky statusbar. I think ideally when the last item is removed from the statusbar, it sets a timer for a second or so in the future and then when that timer fires, if the statusbar is still empty, it gets hidden.

Statusbar showing/hiding is something that we also might consider animating at some point.

### PR DESCRIPTION
**Pull request by jansol**
Fixes #234

### CODE REVIEW

File: XiEditor/StatusBar.swift [hunk 1]
```diff
@@ -210,9 +212,20 @@ class StatusBar: NSView {
         self.needsDisplay = true
     }
 
-    // Hides the status bar if there is no item currently.
-    func checkStatusBarVisibility() {
-        self.isHidden = currentItems.isEmpty
+    // Shows/hides the statusbar based on whether it has any contents
```
Comment (nangtrongvuon) [inline XiEditor/StatusBar.swift]: You should add a period at the end here for punctuation :p

**Comment by jansol** [comment]
(Also renamed the function in question because "check" sounded rather awkward)

File: XiEditor/StatusBar.swift [hunk 2]
```diff
@@ -210,9 +212,20 @@ class StatusBar: NSView {
...
+    func updateStatusBarVisibility() {
+        if !currentItems.isEmpty {
+            self.hideTimer?.invalidate()
+            self.isHidden = false
+        } else if !self.isHidden {
+            // Wait a moment before hiding the bar to avoid blinking in the case
+            // that a single item is added and removed repeatedly in rapid succession.
+            if #available(OSX 10.12, *) {
+                hideTimer = Timer.scheduledTimer(withTimeInterval: 1.0, repeats: false) {_ in self.isHidden = true}
```
Comment (nangtrongvuon) [inline XiEditor/StatusBar.swift]: The block closure at the end should have some spaces to be easier to read, so should look like  `{ _ in self.isHidden = true }`

**Comment by cmyr** [comment]
👍

**Comment by nangtrongvuon** [comment]
Hey, as a last note, can you change `2dcefdb`'s commit message to say:  "Fixed up grammar and punctuation"?

**Comment by cmyr** [comment]
@nangtrongvuon is nittier than me!

**Pull request merged**